app/controllers/likes_controller.rb

```diff
@@ -0,0 +1,24 @@
+class LikesController < ApplicationController
+  before_action :current_user, only: [:create]
+
+  def create
+    @post = Post.find(params[:post_id])
+
+    if already_liked?
+      redirect_to user_post_path(@current_user.id, @post)
+    else
+      @like = @current_user.likes.new
+      @like.user_id = @current_user.id
+      @like.post_id = params[:post_id]
+
+      if @like.save
+        @like.update_likes_counter
+        redirect_to user_post_path(@current_user.id, @post)
+      end
+    end
```

Comment:
- [ ] With this logic, you are complicating the method, I would recommend using another way to implement this logic instead of complicating the logic.

Reply:
- [ ] With this comment, you are complicating everything I suggest you be more specific thank you.